fastd 22 build issue

> clang -Isrc/liblibfastd.a.p -Isrc -I../src -I/opt/homebrew/Cellar/libuecc/7/include/libuecc-7 -I/opt/homebrew/Cellar/json-c/0.15/include -I/opt/homebrew/Cellar/json-c/0.15/include/json-c -I/opt/homebrew/Cellar/openssl@1.1/1.1.1k/include -I/opt/homebrew/Cellar/libsodium/1.0.18_1/include -fcolor-diagnostics -Wall -Winvalid-pch -Wextra -std=c99 -O3 -D_GNU_SOURCE -D__APPLE_USE_RFC_3542 -MD -MQ src/liblibfastd.a.p/handshake.c.o -MF src/liblibfastd.a.p/handshake.c.o.d -o src/liblibfastd.a.p/handshake.c.o -c ../src/handshake.c
../src/handshake.c:293:14: error: implicit declaration of function 'htobe16' is invalid in C99 [-Werror,-Wimplicit-function-declaration]
.length = htobe16(sizeof(header)),
^
1 error generated.

relates to 
- Homebrew/homebrew-core#80211

full build log:
https://github.com/Homebrew/homebrew-core/runs/2939005147

--- a/src/util.h
+++ b/src/util.h
@@ -98,6 +98,18 @@ static inline size_t ssub_size_t(size_t a, size_t b) {
 
 #include <libkern/OSByteOrder.h>
 
+/** Converts a 16bit integer from host byte order to big endian  */
+#define htobe16(x) OSSwapHostToBigInt16(x)
+
+/** Converts a 16bit integer from host byte order to little endian  */
+#define htole16(x) OSSwapHostToLittleInt16(x)
+
+/** Converts a 16bit integer from big endian to host byte order */
+#define be16toh(x) OSSwapBigToHostInt16(x)
+
+/** Converts a 16bit integer from little endian to host byte order */
+#define le16toh(x) OSSwapLittleToHostInt16(x)
+
 /** Converts a 32bit integer from host byte order to big endian  */
 #define htobe32(x) OSSwapHostToBigInt32(x)
 
@@ -111,6 +123,12 @@ static inline size_t ssub_size_t(size_t a, size_t b) {
 #define le32toh(x) OSSwapLittleToHostInt32(x)
 
 #elif !defined(HAVE_LINUX_ENDIAN)
+
+/** Converts a 16bit integer from big endian to host byte order */
+#define be16toh(x) betoh16(x)
+
+/** Converts a 16bit integer from little endian to host byte order */
+#define le16toh(x) letoh16(x)
 
 /** Converts a 32bit integer from big endian to host byte order */
 #define be32toh(x) betoh32(x)

